In [7]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time
import pyautogui




service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")
options.add_argument("headless")
options.add_argument("window-size=1920x1080")
# options.add_argument("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")
# f12 console에 window.navigator.userAgent를 치면 정보를 볼 수 있다. 복붙
options.add_argument("lang=ko_KR")

driver = webdriver.Chrome(service = service, options = options)

driver.get("https://davelee-fun.github.io/")

screen_width, screen_height = pyautogui.size()
driver.set_window_size(screen_width,screen_height)
element = driver.find_element(By.TAG_NAME, "h1")

print(element.text)

time.sleep(3)

driver.quit()

Teddy의 선물 블로그


#### 다음 웹 사이트에 접속해서 selenium을 활용해 크롤링해오세요.
- https://crawlingtest.netlify.app/
- 부모요소의 id = dev_course_list이고 class = course인 데이터만 출력

In [11]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time




service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")
for element in elements :
    print(element.text)

time.sleep(1)
driver.quit()

(초급) - 강사가 실제 사용하는 자동 프로그램 소개 [2]
(초급) - 필요한 프로그램 설치 시연 [5]
(초급) - 데이터를 엑셀 파일로 만들기 [9]
(초급) -     엑셀 파일 이쁘게! 이쁘게! [8]
(초급) -     나대신 주기적으로 파이썬 프로그램 실행하기 [7]
(초급) - 파이썬으로 슬랙(slack) 메신저에 글쓰기 [40]
(초급) - 웹사이트 변경사항 주기적으로 체크해서, 메신저로 알람주기 [12]
(초급) - 네이버 API 사용해서, 블로그에 글쓰기 [42]
(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 커리큘럼 일정표 안에 있는 데이터를 모두 수집해 오세요.

In [14]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time

service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

elements = driver.find_elements(By.CSS_SELECTOR, "tbody tr")
for element in elements :
    print(element.text)
time.sleep(1)
driver.quit()

일정 커리큘럼 타이틀 난이도
5.1 ~ 6.15 나만의 엣지있는 블로그 사이트 만들기 (취미로 익히는 IT) 초급
6.16 ~ 7.31 파이썬과 데이터과학 첫걸음 (IT 기본기 익히기) 중급


#### 아래 데이터만 추출해오세요.
- (중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]

In [18]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time

service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://crawlingtest.netlify.app/")

elements = driver.find_elements(By.CSS_SELECTOR, "ul#dev_course_list > li.course")

print(elements[-1].text)

time.sleep(1)
driver.quit()

(중급) - 자동으로 쿠팡파트너스 API 로 가져온 상품 정보, 네이버 블로그/트위터에 홍보하기 [412]


#### 아래 사이트에서 class값이 news인 값을 추출해오세요.
- https://divjason.github.io/sellenium-test/

In [21]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time

service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

element = driver.find_element(By.CSS_SELECTOR, "span.news")
print(element.text)

time.sleep(1)
driver.quit()

뉴스: 머신러닝 강좌가 나왔어요~~~


#### input 태그 요소에 아이디 error@error.com / 비밀번호 1234 입력 후 로그인 버튼을 자동으로 누르도록 해주세요.

In [28]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time

service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

options.add_argument("--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

ID = driver.find_element(By.CSS_SELECTOR, "input#username")
PW = driver.find_element(By.CSS_SELECTOR, "input#password")
ID.clear()
PW.clear()
ID.send_keys("error@error.com")
PW.send_keys("1234")

time.sleep(5)

LogIn = driver.find_element(By.CSS_SELECTOR, 'input[type="submit"][value="로그인"]')
LogIn.send_keys(Keys.RETURN)


time.sleep(5)
driver.quit()

- 클릭 이후 이동한 페이지에서 message라는 클래스 속성값을 가지고 잇는 요소의 텍스트 정보 출력
- 로그아웃 후 다시 message 출력

In [36]:
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.common.keys import Keys
import time

service = Service("C:\\Users\\user\\Downloads\\chromedriver\\chromedriver.exe")
options = Options()
options.add_argument("--disable-gpu")
options.add_argument("--start-maximized")

options.add_argument("--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36")

driver = webdriver.Chrome(service = service, options = options)
driver.get("https://divjason.github.io/sellenium-test/")

ID = driver.find_element(By.CSS_SELECTOR, "input#username")
PW = driver.find_element(By.CSS_SELECTOR, "input#password")
ID.clear()
PW.clear()
ID.send_keys("errorr@error.com")
PW.send_keys("1234r")

time.sleep(5)

LogIn = driver.find_element(By.CSS_SELECTOR, 'input[type="submit"][value="로그인"]')
LogIn.send_keys(Keys.RETURN)

message = driver.find_element(By.CSS_SELECTOR, "div.message")
print(message.text)
# 클릭 이후 이동한 페이지에서 message라는 클래스 속성값을 가지고 잇는 요소의 텍스트 정보 출력

time.sleep(5)
driver.quit()

실제 사이트는 https://megaitlab.com/ 입니다
본 페이지는 로그인된 페이지입니다!
뉴스: 머신러닝 강좌가 나왔어요~~~
